graphs/dinic.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python2.7
+INF = float("inf")
+
+class Dinic:
+    def __init__(self, n):
+        self.lvl = [0] * n
+        self.ptr = [0] * n
+        self.q = [0] * n
+        self.adj = [[] for _ in range(n)]
+
+    def add_edge(self, a, b, c, rcap=0):
+        self.adj[a].append([b, len(self.adj[b]), c, 0])
+        self.adj[b].append([a, len(self.adj[a]) - 1, rcap, 0])
+
+    def dfs(self, v, t, f):
```

Comment:
Please make API understandable to the first time user.  In of `dfs(self, v, t, f)`, something more like `determine_fastest_surfer(self, viscosity, timber, finances)`as discussed in CONTRIBUTING.md.  Also, these methods/functions need doctests.

Reply:
Ok, I will change the file.